robot/web/web.go

```diff
@@ -365,42 +364,44 @@ func (svc *webService) updateResources(resources map[resource.Name]resource.Reso
 		groupedResources[n.API] = r
 	}
 
-	apiRegs := resource.RegisteredAPIs()
-	for a, v := range groupedResources {
-		apiResColl, ok := svc.services[a]
-		// TODO(RSDK-144): register new service if it doesn't currently exist
+	// All known APIs have pre-registered resource collections, so loop through them. We want to empty out any collections if
+	// there are no longer resources available in that API.
+	for api, coll := range svc.services {
+		group, ok := groupedResources[api]
 		if !ok {
-			reg, ok := apiRegs[a]
-			var apiResColl resource.APIResourceCollection[resource.Resource]
-			if ok {
-				apiResColl = reg.MakeEmptyCollection()
-			} else {
-				// Log a warning here to remind users to register their APIs. Do not warn if the resource is internal to the RDK or
-				// the resource is handled by a remote with a possibly separate API registration. Modular resources will
-				// have API registrations already and should not reach this point in the method.
-				if a.Type.Namespace != resource.APINamespaceRDKInternal {
-					for n := range v {
-						if !n.ContainsRemoteNames() {
-							svc.logger.Warnw(
-								"missing registration for api, resources with this API will be unreachable through a client", "api", n.API)
-							break
-						}
-					}
+			// create an empty map of resources if one does not exist
+			group = make(map[resource.Name]resource.Resource)
+		}
+		if err := coll.ReplaceAll(group); err != nil {
+			return err
+		}
+		delete(groupedResources, api)
+	}
+
+	// If there are any groupedResources remaining, check if they are registered/internal/remote.
+	//  * Custom APIs are registered and do not have a dedicated gRPC service as requests for them are routed through the
+	//    foreignServiceHandler.
+	//  * Internal services do not have an associated gRPC API and so can be safely ignored.
+	//  * Remote resources with unregistered APIs are possibly handled by the remote robot and requests would be routed through the
+	//    foreignServiceHandler.
+	for api, group := range groupedResources {
+		apiRegs := resource.RegisteredAPIs()
+		_, ok := apiRegs[api]
+		if ok {
+			// If registered, the API is most likely a custom API registered through modular resources.
+			continue
+		}
+		// Log a warning here to remind users to register their APIs.
+		if api.Type.Namespace != resource.APINamespaceRDKInternal {
+			for n := range group {
+				if !n.ContainsRemoteNames() {
+					svc.logger.Warnw(
+						"missing registration for api, resources with this API will be unreachable through a client", "api", n.API)
```

Comment:
Could you explain in what scenario we hit this log?

Reply:
it may be possible if you added a resource without registering an api for it - I think the baseremotecontrol service for example would hit this